x/logic/interpreter/fs/parser.go

```diff
@@ -0,0 +1,31 @@
+package fs
+
+import (
+	"context"
+	"fmt"
+	"net/url"
+)
+
+type URIHandler interface {
+	CanOpen(ctx context.Context, uri *url.URL) bool
+	Open(ctx context.Context, uri *url.URL) ([]byte, error)
+}
+
+type Parser struct {
+	Handlers []URIHandler
+}
```

Comment:
Having an array of `URIHandler` let the possibility to configure multiple handlers to resolve the same scheme, maybe we could manage a map of `scheme: handler` instead, the responsibility to identify the proper handler moving to the parser, which may become a router :)

Reply:
I was wondering about this possibility, I choose the solution of an array of handlers because I was considering that the ability of opening an URI can be determined not only on the scheme and the order is determined at the instantiation of handler. 

But I think you're right, and is not necessary to filter more than the scheme, and one scheme is handled by one handler is enough. I will do the modification and transform Parser by a router.